### ISSUE TITLE
[RCCL][gfx90a][Performance] BF16 ReduceScatter AVG is 1.92x slower than SUM at 64 ranks

### ISSUE THREAD

**Issue by JiankunW** [ROCm/rocm-systems#9054]
## Description

BF16 ReduceScatter with `ReduceOp.AVG` is 1.92x slower than the same collective with `ReduceOp.SUM` on a multi-node MI250X/gfx90a cluster.

The difference is reproduced without a model, checkpoint I/O, compute overlap, or profiler.

## Environment

- GPU: AMD Instinct MI250X, gfx90a
- World size: 64 ranks
- PyTorch: 2.10.0+rocm7.1
- ROCm: 7.1.x
- RCCL: 2.27.7

## Results (reproduced by the script below)
Median critical-rank synchronized latency:

BF16 SUM: 8.145 ms
BF16 AVG: 15.618 ms
BF16 AVG/SUM: 1.918x
FP32 SUM: 15.678 ms
FP32 AVG: 15.710 ms
FP32 AVG/SUM: 1.002x
All correctness checks pass.

Impact: PyTorch FSDP2 defaults to ReduceOp.AVG for BF16 gradients. 

Using `module.set_force_sum_reduction_for_comms(True)` changes communication to SUM followed by an equivalent output post-division and restores the expected BF16 speedup.

## Questions

- Is this BF16 AVG/PreMulSum slowdown expected on gfx90a?
- Does this path miss BF16 reduceCopyPacks pipelining?
- Is there a recommended RCCL setting for gfx90a?

### Operating System

SLES 15.7

### CPU

AMD EPYC

### GPU

8 AMD Instinct MI250X nodes (8 GCDs per node)

### ROCm Version

7.1.x

### Steps to Reproduce

```python
import statistics
import time
import torch
import torch.distributed as dist

dist.init_process_group("nccl")
rank = dist.get_rank()
world = dist.get_world_size()
local_rank = int(__import__("os").environ["LOCAL_RANK"])
torch.cuda.set_device(local_rank)
device = torch.device("cuda", local_rank)

full_numel = 314_639_360
assert full_numel % world == 0

def benchmark(dtype, op):
    inp = torch.full(
        (full_numel,), float(rank + 1), dtype=dtype, device=device)
    out = torch.empty(
        (full_numel // world,), dtype=dtype, device=device)

    for _ in range(3):
        dist.reduce_scatter_tensor(out, inp, op=op)
    torch.cuda.synchronize()

    samples = []
    for _ in range(10):
        dist.barrier()
        torch.cuda.synchronize()
        t0 = time.perf_counter()
        dist.reduce_scatter_tensor(out, inp, op=op)
        torch.cuda.synchronize()
        elapsed = (time.perf_counter() - t0) * 1000

        critical = torch.tensor(elapsed, dtype=torch.float64, device=device)
        dist.all_reduce(critical, op=dist.ReduceOp.MAX)
        samples.append(float(critical))

    expected = (
        world * (world + 1) / 2
        if op == dist.ReduceOp.SUM
        else (world + 1) / 2
    )
    assert abs(float(out[0]) - expected) <= max(1.0, expected * 0.02)

    result = statistics.median(samples)
    del inp, out
    torch.cuda.empty_cache()
    return result

for dtype in (torch.bfloat16, torch.float32):
    sum_ms = benchmark(dtype, dist.ReduceOp.SUM)
    avg_ms = benchmark(dtype, dist.ReduceOp.AVG)
    if rank == 0:
        print(dtype, "SUM", sum_ms, "AVG", avg_ms,
              "AVG/SUM", avg_ms / sum_ms)

dist.destroy_process_group()
```

### (Optional for Linux users) Output of /opt/rocm/bin/rocminfo --support

<details>
<summary>rocminfo --support output</summary>

```
Paste output here
```

</details>

**Comment by yousizaitianqiong** [comment]
I would like to contribute a small, test-first first slice for this issue, but want to confirm the intended boundary before writing code.

Proposal:
- work only in `projects/rccl-tests`;
- add an opt-in native `reduce_scatter_perf`/pytest path that pairs BF16 `AVG` and `SUM`, plus FP32 controls, while preserving the default matrix;
- record correctness, selected algorithm/protocol/channel count, and repeatable timing metadata;
- leave RCCL algorithms, public APIs, and the existing default floating-AVG skip unchanged unless maintainers explicitly request a different direction.

The available hardware I can use has a different RCCL/ROCm stack from the issue report, so any result would be clearly labeled as a native RCCL result for that stack, not a reproduction claim for ROCm 7.1/PyTorch.

Could you please confirm:
1. Is `projects/rccl-tests` the preferred location for an opt-in native test/harness?
2. Which RCCL version(s), GPU topology, and MPI launcher should be treated as the minimum supported validation target?
3. Would manual hardware evidence plus focused unit/pytest coverage be useful for an initial PR, or should the first contribution be documentation only?
4. Should floating BF16 AVG remain opt-in until ReduceScatter parity work is completed?

I will wait for scope confirmation before cloning, creating a branch, or proposing a PR.
